internal/versions/versions.go

```diff
@@ -175,42 +171,88 @@ func (h *handler) GetDefaultReleaseImage(cpuArchitecture string) (*models.Releas
 
 // Returns the OsImage entity
 func (h *handler) GetOsImage(openshiftVersion, cpuArchitecture string) (*models.OsImage, error) {
-	versionKey, err := h.getKey(openshiftVersion)
-	if err != nil {
-		return nil, err
-	}
-
-	for _, osImage := range h.osImages {
+	// Find OS images of relevant CPU architecture
+	osImages := funk.Filter(h.osImages, func(osImage *models.OsImage) bool {
+		// Empty implies default CPU architecture
+		if swag.StringValue(osImage.CPUArchitecture) == "" {
+			return cpuArchitecture == "" || cpuArchitecture == common.DefaultCPUArchitecture
+		}
 		if cpuArchitecture == "" {
-			// Empty implies default CPU architecture
 			cpuArchitecture = common.DefaultCPUArchitecture
 		}
-		if *osImage.OpenshiftVersion == versionKey && *osImage.CPUArchitecture == cpuArchitecture {
-			return osImage, nil
-		}
+		return swag.StringValue(osImage.CPUArchitecture) == cpuArchitecture
+	})
+	if funk.IsEmpty(osImages) {
+		return nil, errors.Errorf("The requested CPU architecture (%s) isn't specified in OS images list", cpuArchitecture)
```

Comment:
Is this particularly important to report?
Could we just check both requirements in the same `Find` instead of filtering first?

Reply:
Well I've separated the architecture filtering just to avoid checking it again later in the func. I.e. if a specific x.y.z isn't found, we fallback to x.y, etc. So seems simpler to check the architecture only once. We could still avoid the early return here if it would be clearer :/

Reply:
Nah this is fine. Things look separated out pretty clearly now.